### PR TITLE
add test to vision base

### LINKED ISSUE #957: [BUG] No tests for vision suite and most stuff in vision base

### PR DESCRIPTION
**Pull request by benisraeldan**
#### Reference Issues/PRs

Resolves #957 

#### What does this implement/fix? Explain your changes.
Adds tests to vision base

#### Any other comments?
Most code is covered by testing implemented checks, but base test can help identify specific issues. 

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: tests/vision/base/test_base_checks.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            executions["compute"] += 1
+            return CheckResult(None)
+
+    # Act
+    DummyCheck().run(coco_dataset)
+    DummyTrainTestCheck().run(coco_dataset, coco_dataset)
+    DummyModelCheck().run(coco_model)
+
+    # Assert
+    assert_that(executions, is_({"initialize_run": 3, "compute": 3, "update": 6}))
```
Comment (matanper) [inline tests/vision/base/test_base_checks.py:82]: the update shouldn't be 4?

File: tests/vision/base/test_suite.py [hunk 2]
```diff
@@ -21,36 +22,184 @@
...
+    suite = Suite("test",
+                  DummyTrainTestCheck())
+    assert_that(calling(suite.run).with_args(test_dataset=coco_dataset),
+                raises(DatasetValidationError,
+                       "Can't initialize context with only test. if you have single dataset, initialize it as train"))
+
+    assert_that(executions, is_({}))
+
+
+def test_suite_execution_with_missing_train():
```
Comment (matanper) [inline tests/vision/base/test_suite.py:168]: same function name twice

File: tests/vision/base/test_vision_data.py [hunk 3]
```diff
@@ -293,8 +293,80 @@ def test_data_at_batch_of_index(mnist_dataset_train):
...
+                return [8]
+            elif batch =='bad tensor':
+                return [torch.Tensor([])]
+            elif batch == 'short tensor':
+                return [torch.Tensor([[1,2], [1,2]])]
+
+            raise DeepchecksNotImplementedError('infer_on_batch() must be implemented in a subclass')
+
+    detection_data = DummyDetectionData(coco_dataset)
+    assert_that(calling(detection_data.validate_label).with_args('not_list'),
```
Comment (matanper) [inline tests/vision/base/test_vision_data.py:341]: why not just pass the value itself as batch instead of this if-else?
```
with_args(7)
```

**Review by matanper — APPROVED** [review]

File: tests/vision/base/test_suite.py [hunk 4]
```diff
@@ -21,36 +22,184 @@
...
+            executions["compute"] += 1
+            raise DeepchecksValueError('bad compute')
+            return CheckResult(None)
+
+    suite = Suite("test",
+                  DummyCheck(),
+                  DummyTrainTestCheck())
+    suite.run(train_dataset=coco_dataset, test_dataset=coco_dataset)
+
+    assert_that(executions, is_({'initialize_run': 3, 'update': 8, 'compute': 3}))
```
Comment (JKL98ISR) [inline tests/vision/base/test_suite.py:140]: can you assert you have the exception in the results?

**Review by JKL98ISR — CHANGES_REQUESTED** [review]
just add some checks for suite results i.e exceptions and output

**Comment by JKL98ISR** [comment]
can you also validate other exceptions? I see many that aren't covered. we can easily reach 100% coverage there

**Pull request merged**